run-checks

```diff
@@ -140,10 +142,14 @@ if [ "$UNIT" = 1 ]; then
     echo Running tests from $(pwd)
     for pkg in $(go list ./... | grep -v '/vendor/' ); do
         go test -i $pkg
-        $goctest -v -coverprofile=.coverage/profile.out $pkg
+        $goctest -v -coverprofile=.coverage/profile.out -covermode=atomic $pkg
         append_coverage .coverage/profile.out
     done
 
+    # upload to codecov.io if on travis
+    if [ "${TRAVIS_BUILD_NUMBER:-}" ]; then
+        bash <(curl -s https://codecov.io/bash) -f .coverage/coverage.out
```

Comment:
woa! (I know we don't have commit bits inside travis so should be ok but looks a bit scary anyway)